dev-support/jenkins-scripts/generate-hbase-website.sh

```diff
@@ -23,73 +23,173 @@
-rm -Rf -- *.patch *.patch.zip hbase/target target *.txt hbase-site
+set -e
+function usage {
+  echo "Usage: ${0} [options] /path/to/hbase/checkout"
+  echo ""
+  echo "    --working-dir /path/to/use  Path for writing logs and a local checkout of hbase-site repo."
+  echo "                                if given must exist."
+  echo "                                defaults to making a directory in /tmp."
+  echo "    --local-repo /path/for/maven/.m2  Path for putting local maven repo."
+  echo "                                if given must exist."
+  echo "                                defaults to making a clean directory in --working-dir."
+  echo "    --publish                   if given, will attempt to push results back to the hbase-site repo."
+  echo "    --help                      show this usage message."
+  exit 1
+}
+# if no args specified, show usage
+if [ $# -lt 1 ]; then
+  usage
+fi
+
+# Get arguments
+declare component_dir
+declare working_dir
+declare local_repo
+declare publish
+while [ $# -gt 0 ]
+do
+  case "$1" in
+    --working-dir) shift; working_dir=$1; shift;;
+    --local-repo) shift; local_repo=$1; shift;;
+    --publish) shift; publish="true";;
+    --) shift; break;;
+    -*) usage ;;
+    *)  break;;  # terminate while loop
+  esac
+done
+
+# should still have where component checkout is.
+if [ $# -lt 1 ]; then
+  usage
+fi
+component_dir="$(cd "$(dirname "$1")"; pwd)/$(basename "$1")"
+
+if [ -z "${working_dir}" ]; then
+  working_dir=/tmp
+  while [[ -e ${working_dir} ]]; do
+    working_dir=/tmp/hbase-generate-website-${RANDOM}.${RANDOM}
+  done
+  mkdir "${working_dir}"
+else
+  # absolutes please
+  working_dir="$(cd "$(dirname "${working_dir}")"; pwd)/$(basename "${working_dir}")"
+  if [ ! -d "${working_dir}" ]; then
+    echo "passed working directory '${working_dir}' must already exist."
+    exit 1
+  fi
+fi
+
+echo "You'll find logs and temp files in ${working_dir}"
+
+if [ -z "${local_repo}" ]; then
+  local_repo="${working_dir}/.m2/repo"
+  # Nuke the local maven repo each time, to start with a known environment
+  rm -Rf "${local_repo}"
+  mkdir -p "${local_repo}"
+else
+  # absolutes please
+  local_repo="$(cd "$(dirname "${local_repo}")"; pwd)/$(basename "${local_repo}")"
+  if [ ! -d "${local_repo}" ]; then
+    echo "passed directory for storing the maven repo  '${local_repo}' must already exist."
+    exit 1
+  fi
+fi
 
 # Set up the environment
-export JAVA_HOME=$JDK_1_8_LATEST__HOME
-export PATH=$JAVA_HOME/bin:$MAVEN_3_3_3_HOME/bin:$PATH
-export MAVEN_OPTS="-XX:MaxPermSize=256m -Dmaven.repo.local=${LOCAL_REPO}"
+if [ -z "${JAVA_HOME}" ]; then
+  JAVA_HOME="${JDK_1_8_LATEST__HOME}"
+  export JAVA_HOME
+  export PATH="${JAVA_HOME}/bin:${PATH}"
+fi
+if [ -z "${MAVEN_HOME}" ]; then
+  MAVEN_HOME="${MAVEN_3_3_3_HOME}"
+  export MAVEN_HOME
+  export PATH="${MAVEN_HOME}/bin:${PATH}"
+fi
+export MAVEN_OPTS="-XX:MaxPermSize=256m -Dmaven.repo.local=${local_repo}"
```

Comment:
I think the MaxPermSize is deprecated and the script bitches about it each time it runs.

Reply:
so don't set it at all or try to check the java version or something?

Reply:
Apparently not needed anymore: http://openjdk.java.net/jeps/122

Reply:
do we only ever build the website with the master branch? If so then I think we're good since that's always built with java 8.

Reply:
Correct. We probably should review that because I think it means that the other branches don’t have up to date Javadoc available anywhere for browsing, but that’s out of scope for this.

Reply:
FWIW, when I do a branch-1.2 release I just run `mvn javadoc:aggregate` to get the javadocs to copy in place.